### PR TITLE
build: dependency updates

### PR DESCRIPTION
**Pull request by DanStough**
This updates all the dependencies to latest. Some notes:
* The AWS SDK changed semantic. I think most of the changes are straightforward, but I used the docs for [pagination](https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#using-paginators) and [errors](https://aws.github.io/aws-sdk-go-v2/docs/handling-errors/) to know what to change. 
* I copied over the relevant files we were importing from the root Consul module. We stopped editing the `go.mod` to do root imports, so this will prevent some weirdness with the dependency on the `api` module.
* I changed some of the names to be more intuitive, but I'm not married to anything.
* The only dependency that didn't really update was armon/go-metrics. It's pulled in indirectly through consul/api -> serf. It seems like we're going to have to figure this out sooner than later or it will be a problem.

The following will be added in future PRs:
* GHA for build, test, security scan, and changelog checker.
* CRT pipeline

### CODE REVIEW

File: catalog/aws.go [hunk 1]
```diff
@@ -204,57 +209,44 @@ func (a *aws) rekeyHealths(name string, healths map[string]health) map[string]he
...
-func statusToCustomHealth(h health) sd.CustomHealthStatus {
-	var result sd.CustomHealthStatus
-	switch h {
-	case passing:
-		result = sd.CustomHealthStatusHealthy
-	case critical:
-		result = sd.CustomHealthStatusUnhealthy
-	}
-	return result
-}
```
Comment (DanStough) [inline catalog/aws.go:229]: Unused

File: catalog/aws_test.go [hunk 2]
```diff
@@ -6,32 +6,19 @@ package catalog
...
-	sdi.ServiceDiscoveryAPI
-}
-
-func (m *mSDClint) CreateServiceRequest(input *sd.CreateServiceRequest) (*sd.CreateServiceOutput, error) {
-	return nil, nil
-}
-
-func (m *mSDClint) RegisterInstanceRequest(input *sd.RegisterInstanceRequest) (*sd.RegisterInstanceOutput, error) {
-	return nil, nil
-}
```
Comment (DanStough) [inline catalog/aws_test.go:25]: Also unused

File: Makefile [hunk 3]
```diff
@@ -118,5 +120,18 @@ clean:
...
+# We support all the same CLI flags for connecting to Consul as the main CLI.
+# Rather than import the root Consul module, which is not supported, we copy the
+# relevant module. This is similar to what we do in consul-dataplane.
+# We don't include config_test.go because it pulls in a directory with test data.
+.PHONY: copy-http-flags
+copy-http-flags: ## copy http flags modules from consul
+	for file in config.go http.go http_test.go merge.go usage.go; do \
+		curl --fail https://raw.githubusercontent.com/hashicorp/consul/main/command/flags/$$file | \
+		sed 's/BUSL-1.1/MPL-2.0/' | \
+		sed '1s:^:// Code generated by make copy-http-flags. DO NOT EDIT.\n:' | \
```
Comment (zalimeni) [inline Makefile:132]: 👍🏻 given the current options and level of investment in this repo, nice solution.

File: catalog/sync_test.go [hunk 4]
```diff
@@ -101,13 +105,16 @@ func runSyncTest(t *testing.T, namespaceID string) {
...
 	err = deleteServiceInAWS(a, aID)
 	if err != nil {
-		t.Logf("error deleting service: %s", err)
+		t.Logf("error deleting service in AWS: %s", err)
+	}
+	err = deleteServiceInConsul(c, cID)
+	if err != nil {
+		t.Logf("error deleting service in Consul: %s", err)
 	}
-	deleteServiceInConsul(c, cID)
```
Comment (zalimeni) [inline catalog/sync_test.go:110]: 👍🏻

File: go.mod [hunk 5]
```diff
@@ -1,46 +1,60 @@
 module github.com/hashicorp/consul-aws
 
-go 1.14
+go 1.22.1
```
Comment (zalimeni) [inline go.mod]: We got some CVE patches in `1.22.2` this week, so suggesting we bump to that
```suggestion
go 1.22.2
```

**Review by zalimeni — APPROVED** [review]
One minor suggestion, but otherwise LGTM - thank you for tackling this!

**Pull request merged**